portal/server/helpers/addBusinessDays.js

```diff
@@ -0,0 +1,23 @@
+const { addBusinessDays, isSameDay } = require('date-fns');
+
+/**
+ * Adds the number of days given to the starting date provided excluding
+ * weekends and holidays, if provided.
+ * @param {Date} startingDate - The date value to start from.
+ * @param {Integer} numberOfDays - The number of business days to increment by.
+ * @param {Array} holidays - A list of dates which should be excluded as holidays.
+ * @returns {Date} - The final date value.
+ */
+
+const addBusinessDaysWithHolidays = (startingDate, numberOfDays, holidays) => {
+  let result = startingDate;
+  for (let i = 0; i < numberOfDays; i += 1) {
+    let temporaryDate = addBusinessDays(result, 1);
+    const isHoliday = holidays.some((holiday) => isSameDay(temporaryDate, holiday));
+    if (isHoliday) { temporaryDate = addBusinessDays(temporaryDate, 1); }
+    result = temporaryDate;
+  }
```

Comment:
I found a possibly bug in this algorithm (though unlikely to come up). Imagine You start on Monday, but both Tuesday and Wednesday are bank holidays. I think that, when this algorithm reaches `i = 1` (ie. `result` should be Thursday as that's one business day including both bank holidays) `result` will actually be Wednesday. The way I think you can get around this is replacing the `if` block with:
```
while (isHoliday) {
    temporaryDate = addBusinessDays(temporaryDate, 1);
    isHoliday = holidays.some(...);
}
```
You will also have to declare `isHoliday` as `let`